util/dvsim/verilator-report-parser.py

```diff
@@ -0,0 +1,93 @@
+        .hjson file with lint messages and their severities.
+
+        The script returns nonzero status if any warnings or errors are
+        present.
+        """)
+    parser.add_argument('--repfile',
+                        type=lambda p: Path(p).resolve(),
+                        default="./verilator.log'",
+                        help="""The script searches the log file provided.
+                        Defaults to './verilator.log'""")
+
+    parser.add_argument('--outfile',
+                        type=lambda p: Path(p).resolve(),
+                        default="./results.hjson",
+                        help="""Path to the results Hjson file.
+                        Defaults to './results.hjson'""")
+
+    args = parser.parse_args()
+
+    # Patterns for lint.log
+    parser_args = {
+        args.repfile: [
+            # If lint warnings have been found, the lint tool will exit
+            # with a nonzero status code and fusesoc will always spit out
+            # an error like
+            #
+            #    ERROR: Failed to build ip:core:name:0.1 : 'make' exited with an error code
+            #
+            # If we found any other warnings or errors, there's no point in
+            # listing this too. BUT we want to make sure we *do* see this
+            # error if there are no other errors or warnings, since that
+            # shows something has come unstuck. (Probably the lint tool
+            # spat out a warning that we don't understand)
+            ("fusesoc-error",
+             r"^ERROR: Failed to build .* 'make' exited with an error code"),
+            ("errors",
+             r"^(?!ERROR: Failed to build .* 'make' exited with an error code)ERROR: .*"
+             ),
+            ("errors",
+             # This is a redundant Verilator error that we ignore, since we
+             # already parse out each individual error.
+             r"^(?!%Error: Exiting due to .* warning.*)%Error: .*"),
+            # TODO(https://github.com/olofk/edalize/issues/90):
+            # this is a workaround until we actually have native Edalize
+            # support for JasperGold and "formal" targets
+            ("warnings",
+             r"^(?!WARNING: Unknown item formal in section Target)WARNING: .*"),
+            ("warnings", r"^%Warning: .* "),
+            ("lint_errors", r"^%Error-.*"),
+            ("lint_warnings", r"^%Warning-.*")
+        ]
+    }
+
+    # Parse logs
+    parser = LintParser()
+    num_messages = parser.get_results(parser_args)
+
+    # Write out results file
+    parser.write_results_as_hjson(args.outfile)
+
+    # return nonzero status if any warnings or errors are present
+    # lint infos do not count as failures
+    if num_messages['error'] > 0 or num_messages['warning'] > 0:
+        log.info("Found %d lint errors and %d lint warnings",
+                 num_messages['error'],
+                 num_messages['warning'])
+        sys.exit(1)
```

Comment:
Ideally, dvsim would decide whether to exit with errors.

Reply:
Yes this will go away once the parser is merged into the Dvsim core code.